plugin/formatting.py

```diff
@@ -27,7 +28,8 @@ def on_pre_save(self) -> None:
             self._purge_changes_if_needed()
             self._will_save_wait_until(session)
 
-        if self.view.settings().get("lsp_format_on_save"):
+        format_on_save = self.view.settings().get("lsp_format_on_save", None)
+        if (format_on_save is not None and format_on_save) or settings.lsp_format_on_save:
```

Comment:
Actually this is not correct logic. View setting should override global always, regardless if true or false.